mirrord-agent/src/main.rs

```diff
@@ -104,13 +108,79 @@ pub struct PeerMessage {
     peer_id: PeerID,
 }
 
+/// Helper function that loads the process' environment variables, and selects only those that were
+/// requested from `mirrord-layer`.
+///
+/// Returns an error if none of the requested environment variables were found.
+async fn select_env_vars(
+    environ_path: PathBuf,
+    override_env_vars: HashMap<String, String>,
+) -> Result<HashMap<String, String>, ResponseError> {
+    debug!(
+        "select_env_vars -> environ_path {:#?} override_env_vars {:#?}",
+        environ_path, override_env_vars
+    );
+
+    let mut environ_file = tokio::fs::File::open(environ_path).await.map_err(|fail| {
+        ResponseError::FileOperation(FileError {
```

Comment:
erm leveraging the file operation errors seems a bit shady?

Reply:
But this fails in a file operation though.

Reply:
Indeed but layer sent a env request but it's getting a file operation error.. we can discuss whether errors/request reflect lower level implementation or high level logic. IMO it's high level (i.e layer doesn't care what failed exactly, only that the request it had sent has failed)
but it doesn't matter too much :)

Reply:
Yeah both approaches make total sense, and it is indeed a discussion between if we want layer to have a lower or higher level view of what happens in agent.

I would vote for lower level, but I'm game for either. 

So how should I proceed?

Reply:
I think calling it "EnvGetError" or a like is better, but wouldn't waste too much time on it atm.